File: ruby/algo_ladder/flip_hash/flip_hash.rb
# Given a hash, create a new hash that has the keys and values switched.

# Input: {"a" => 1, "b" => 2, "c" => 3}
# Output: {1 => "a", 2 => "b", 3 => "c"}

def flip_hash (input_hash)
  result = {}
  input_hash.each do |key, value|
    result[value] = key
  end
  return result
end

p flip_hash({"a" => 1, "b" => 2, "c" => 3})

File: javascript/algoLadder/flipHash/flipHash.js
// Given a hash, create a new hash that has the keys and values switched.

// Input: {"a" => 1, "b" => 2, "c" => 3}
// Output: {1 => "a", 2 => "b", 3 => "c"}

const flipHash = function (inputHash) {
  let result = {}

  for (let key in inputHash) {
    result[inputHash[key]] = key
  }
  
  return result
}

console.log(flipHash({"a": 1, "b": 2, "c": 3}))